zip-0230.rst

```diff
@@ -4,11 +4,347 @@
+
+Orchard-ZSA Asset Burn Description
+----------------------------------
+
+An Orchard-ZSA Asset Burn description is encoded in a transaction as an instance of an ``AssetBurn`` type:
+
++-------+---------------+-----------------------------+---------------------------------------------------------------------------------------------------------------------------+
+| Bytes | Name          | Data Type                   | Description                                                                                                               |
++=======+===============+=============================+===========================================================================================================================+
+| 32    | ``AssetBase`` | ``byte[32]``                | For the Orchard-based ZSA protocol, this is the encoding of the Asset Base :math:`\mathsf{AssetBase}^{\mathsf{Orchard}}`. |
++-------+---------------+-----------------------------+---------------------------------------------------------------------------------------------------------------------------+
+| 8     | ``valueBurn`` | :math:`\{1 .. 2^{64} - 1\}` | The amount being burnt.                                                                                                   |
++-------+---------------+-----------------------------+---------------------------------------------------------------------------------------------------------------------------+
+
+The encodings of each of these elements are defined in ZIP 226 [#zip-0226]_.
+
+Issuance Action Description (``IssueAction``)
+---------------------------------------------
+
+An issuance action, ``IssueAction``, is the instance of issuing a specific Custom Asset, and contains the following fields:
+
++-----------------------------+--------------------------+-------------------------------------------+---------------------------------------------------------------------+
+| Bytes                       | Name                     | Data Type                                 | Description                                                         |
++=============================+==========================+===========================================+=====================================================================+
+|``2``                        |``assetDescSize``         |``byte``                                   |The length of the asset description string in bytes.                 |  
++-----------------------------+--------------------------+-------------------------------------------+---------------------------------------------------------------------+
+|``assetDescSize``            |``asset_desc``            |``byte[assetDescSize]``                    |A byte sequence of length ``assetDescSize`` bytes which SHOULD be a  |
+|                             |                          |                                           |well-formed UTF-8 code unit sequence according to Unicode 15.0.0     |
```

Comment:
This being a SHOULD is correct (because we don't want consensus to depend on the definition of UTF-8), but it means that the behaviour when it is not a well-formed UTF-8 code sequence needs to be specified.

Reply:
Added behaviour for this scenario -- namely that the wallet provide a warning that the description is not a valid UTF-8 encoding, and then proceed using the bytestring.